File: EProcurement.Frontend/src/types/permission.ts
export interface Permission { 
    id: string; 
    label: string; 
    category: string 
}

export interface PermissionFromApi {
  permissionID: number;
  code: string;
  name: string;
  description: string | null;
  permissionCategoryID: number;
  category: string;
  isActive: boolean;
  createdAt: string;
  createdBy: string;
  updatedAt: string | null;
  updatedBy: string | null;
  deletedAt: string | null;
  deletedBy: string | null;
}

File: EProcurement.Api/Entities/Permission.cs
namespace EProcurement.Api.Entities
{
    public class Permission
    {
        public int PermissionID { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int PermissionCategoryID { get; set; }
        public string PermissionCategoryName { get; set; }
        public bool IsActive { get; set; }

        public DateTime Created_At { get; set; }
        public string Created_By { get; set; }
    }
}
